.github/workflows/build.yaml

```diff
@@ -56,33 +73,100 @@ jobs:
       #
       # Build binary artifacts for Darwin and Linux
       #
-      # TODO: would save time if we could cache go module dependencies across build steps
       - name: Build linux binary distribution
         uses: ./.github/actions/make
         with:
           target: "release"
           os: linux
           arch: amd64
           version: ${{ steps.tag.outputs.tag }}
-      - name: Build darwin binary distribution
+      - name: Build amd64 darwin binary distribution
         uses: ./.github/actions/make
         with:
           target: "release"
           os: darwin
           arch: amd64
           version: ${{ steps.tag.outputs.tag }}
-      - name: Build M1 darwin binary distribution
+
+      - name: Build arm64 M1 darwin binary distribution
         uses: ./.github/actions/make
         with:
           target: "release"
           os: darwin
           arch: arm64
           version: ${{ steps.tag.outputs.tag }}
+
+  sign-and-notarize:
+    runs-on: macos-latest
+    needs: build
+    steps:
+      - name: Checkout current code
+        uses: actions/checkout@v2
+        with:
+          token: ${{ secrets.BROADBOT_TOKEN }}
+
+      - name: Pull cache
+        id: pull-cache
+        uses: actions/cache@v3
+        env:
+          cache-name: build-dir-cache
+        with:
+          path: ./output
+          key: commit-${{ github.sha }}-job-${{ github.run_id }}
+
+      # A second cache is needed to force GH to upload the new macOS tarballs,
```

Comment:
maybe call the two "caches" `unsigned-build-artifacts` and `signed-build-artifacts` for clarity? IDK, it's a matter of taste but that would have helped me personally understand sooner 🤷  (Since caching is not what we're doing as much as copying artifacts across jobs)

Reply:
In the time since this comment, all of the intermediary caches have been given platform-specific names and only the final release cache is called `release-cache`. I've also added a big comment to the top of this file explaining how the caches work for future reference. I think that should do it.